### ISSUE TITLE
[Idea] Add format by resource

### ISSUE THREAD

**Issue by vincentchalamon** [api-platform/core#1778]
Currently, default supported formats are `jsonld`, `json` & `html`. If I want to add a new format, I must add it to api_platform configuration as following:
```yaml
api_platform:
    formats:
        # ...
        pdf:
            mime_types: ['...']
```
But this format will be available for all entry-points of my API. I may want to use this format just for some entry-points.

It should be very useful to add a `format` option directly in the ApiResource annotation, which could be overriden by operation. For instance:
```php
/**
 * @ApiResource(attributes={
 *     "formats"={"jsonld", "json", "html", "csv"={"mime_types"={"text/csv"}}}
 * }, itemOperations={
 *     "put", "delete",
 *     "get"={"formats"={"jsonld", "json", "html", "pdf"={"mime_types"={"application/pdf"}}}}
 * })
 */
class Foo
```
In this example, `csv` format is available in all operations except getItem, and this one has `pdf` format. Those extra formats are only available in this ApiResource.

What do you think of this idea @api-platform/core-team?

**Comment by Toflar** [comment]
I would need that too, should be pretty straight forward to add. Are you creating a PR?

**Comment by vincentchalamon** [comment]
I'm looking for a time-machine to get enough time to work on it @Toflar :wink:

**Comment by meyerbaptiste** [comment]
Don't waste your time looking, create the PR instead! 😄

**Comment by vincentchalamon** [comment]
@Simperfit this has been released in 2.3.0 and can be closed
